commons/all/src/main/java/org/infinispan/commons/util/ConcurrentSmallIntSet.java

```diff
@@ -58,6 +60,33 @@ public ConcurrentSmallIntSet(int maxCapacityExclusive) {
       array = new AtomicIntegerArray(intLength);
    }
 
+   @ProtoFactory
+   static ConcurrentSmallIntSet protoFactory(int[] entries) {
+      int arrayLength = entries.length;
+      ConcurrentSmallIntSet intSet = new ConcurrentSmallIntSet(arrayLength << ADDRESS_BITS_PER_INT);
+
+      int size = 0;
+      for (int i = 0; i < arrayLength - 1; ++i) {
+         int value = entries[i];
+         // Use lazy set - we use set below on the last
+         intSet.array.lazySet(i, value);
+         size += Integer.bitCount(value);
+      }
+      int lastValue = entries[arrayLength - 1];
+      intSet.array.set(arrayLength - 1, lastValue);
+      size += Integer.bitCount(lastValue);
+      intSet.currentSize.addAndGet(size);
+
+      return intSet;
+   }
+
+   @ProtoField(1)
+   int[] entries() {
+      int[] entries = new int[array.length()];
+      for (int i = 0; i < array.length(); i++) entries[i] = array.get(i);
+      return entries;
+   }
+
```

Comment:
How do you feel about running more JMH performance tests? 
Using the new Protostream feature, `IntStream.range(0, array.length()).map(array::get);` may be better (or worse) 
Another alternative is to create an `Iterable` to iterate over `array` elements.

Reply:
I think `Iterable` makes the most sense here, as we can then pass a `List` to the `@ProtoFactory` annotation so that we know the total number of elements and can size the `ConcurrentSmallIntSet` accordingly